pick.c

```diff
@@ -928,16 +930,20 @@ get_key(char *buf, size_t size, size_t *nread)
 		KEY(RIGHT,	"\006"),
 		KEY(UNKNOWN,	NULL),
 	};
-	int	c, i;
+	static unsigned char	buf[8];
+	size_t			len;
+	int			c, i;
 
-	*nread = 0;
+	*key = (const char *)buf;
+	len = 0;
 
 	/*
 	 * Allow SIGWINCH on the first read. If the signal is received, return
 	 * CTRL_L which will trigger a resize.
 	 */
 	toggle_sigwinch(1);
-	buf[(*nread)++] = tty_getc();
+	buf[len++] = tty_getc();
+	buf[len] = '\0';
```

Comment:
How about using `memset` at the beginning to zero out the buffer?

Reply:
Interesting, I wonder if Coverity will complain though: 3e04320.

Reply:
```diff
diff --git a/pick.c b/pick.c
index 73cc0e4..194db1d 100644
--- a/pick.c
+++ b/pick.c
@@ -930,7 +930,7 @@ get_key(const char **key)
                KEY(RIGHT,      "\006"),
                KEY(UNKNOWN,    NULL),
        };
-       static unsigned char    buf[8];
+       static unsigned char    buf[8] = {'\0'};
        size_t                  len;
        int                     c, i;
 
@@ -943,7 +943,6 @@ get_key(const char **key)
         */
        toggle_sigwinch(1);
        buf[len++] = tty_getc();
-       buf[len] = '\0';
        toggle_sigwinch(0);
        if (gotsigwinch) {
                gotsigwinch = 0;
@@ -971,7 +970,6 @@ get_key(const char **key)
                if (len == sizeof(buf) - 1)
                        break;
                buf[len++] = tty_getc();
-               buf[len] = '\0';
        }
 
        if (len > 1 && buf[0] == '\033' && (buf[1] == '[' || buf[1] == 'O')) {
@@ -1006,7 +1004,6 @@ get_key(const char **key)
                        return UNKNOWN;
 
                buf[len++] = tty_getc();
-               buf[len] = '\0';
        }
 
        return PRINTABLE;
```

Reply:
```diff
diff --git a/pick.c b/pick.c
index 73cc0e4..5354ea2 100644
--- a/pick.c
+++ b/pick.c
@@ -937,13 +937,14 @@ get_key(const char **key)
        *key = (const char *)buf;
        len = 0;
 
+       memset(buf, '\0', sizeof(buf));
+
        /*
         * Allow SIGWINCH on the first read. If the signal is received, return
         * CTRL_L which will trigger a resize.
         */
        toggle_sigwinch(1);
        buf[len++] = tty_getc();
-       buf[len] = '\0';
        toggle_sigwinch(0);
        if (gotsigwinch) {
                gotsigwinch = 0;
@@ -971,7 +972,6 @@ get_key(const char **key)
                if (len == sizeof(buf) - 1)
                        break;
                buf[len++] = tty_getc();
-               buf[len] = '\0';
        }
 
        if (len > 1 && buf[0] == '\033' && (buf[1] == '[' || buf[1] == 'O')) {
@@ -1006,7 +1006,6 @@ get_key(const char **key)
                        return UNKNOWN;
 
                buf[len++] = tty_getc();
-               buf[len] = '\0';
        }
 
        return PRINTABLE;
```

Reply:
Coverity didn't complain so this change is in, thanks!